k9mail/src/main/java/com/fsck/k9/ui/messageview/MessageTopView.java

```diff
@@ -300,6 +306,19 @@ private boolean shouldShowPicturesFromSender(ShowPictures showPicturesSetting, M
         return contacts.isInContacts(senderEmailAddress);
     }
 
+    private boolean shouldShowPicturesInCurrentNetwork(boolean onlyShowViaWiFi) {
+        if (!onlyShowViaWiFi) {
+            return true;
+        }
+
+        final ConnectivityManager connectivityManager =
+                (ConnectivityManager) getContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+        NetworkType networkType =
+                NetworkType.fromConnectivityManagerType(connectivityManager.getActiveNetworkInfo().getType());
+
+        return networkType == NetworkType.WIFI;
```

Comment:
Despite the name, this should probably be [isActiveNetworkMetered](https://developer.android.com/reference/android/net/ConnectivityManager.html#isActiveNetworkMetered), which is what 99% of people mean when they say "only on wifi". @cketti ?